### DISCUSSION TITLE
`export prerender=false` with `output: 'static'`

### DISCUSSION THREAD

**Discussion by zadeviggers** [withastro/roadmap#455] [Proposal]
After reading the prerendering RFC, one use-case that seems common (and is mentioned multiple times in the RFC) is a fully static site with a few server-side api routes.
However, the current implementation requires that I set `output: 'server'` in the config, and add `export const prerender = true` on every single static page, which is a lot of extra work and creates the  potential for forgoetting to set a page to be prerendered, and accidentally rendering a high-traffic static page on each request, which could be costly.

Intuitively for this situation, I would set `output: 'static'` in the config, then in the api routes, `export const prerender = false`.

Is there a possibility of introducing this sort of syntax in the future?

**Comment by natemoo-re** [comment]
Hey @zadeviggers! Thanks for the proposal. We have seen a lot of feedback like this around the defaults for the `prerender` feature, so this is something we're actively looking into fixing in a minor release of Astro 2.

I'm not sure if we'll be able to support this feature as a _non-breaking_ change to our existing `output: 'static'` mode. However, we're leaning towards introducing a third `output: 'hybrid'` option that acts exactly as you suggest. Having an explicit opt-in to this behavior means that integrations can update incrementally and this could actually be a non-breaking change.

**Comment by Princesseuh** [comment]
This proposal has been moved to Stage 2! Thank you everyone for voting and contributing. https://github.com/withastro/roadmap/issues/539
